src/samples/coap/iotivity-test-client.c

```diff
@@ -478,6 +478,18 @@ found_resource(struct sol_oic_client *cli, struct sol_oic_resource *res, void *d
     bool (*fill_repr_map)(void *data, struct sol_oic_map_writer *repr_map) = NULL;
     struct sol_str_slice href;
 
+    if (!res)
+        return false;
+
+#ifndef SOL_NO_API_VERSION
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file